pkg/lease/lease.go

```diff
@@ -13,25 +13,48 @@ import (
 
 // LeaseReconciler reconciles a Secret object
 type LeaseReconciler struct {
+	HubKubeClient        kubernetes.Interface
 	KubeClient           kubernetes.Interface
 	LeaseName            string
 	LeaseDurationSeconds int32
+	ClusterName          string
 	componentNamespace   string
 }
 
 func (r *LeaseReconciler) Reconcile() {
 	if len(r.componentNamespace) == 0 {
 		r.componentNamespace = getPodNamespace()
 	}
-	lease, err := r.KubeClient.CoordinationV1().Leases(r.componentNamespace).Get(r.LeaseName, metav1.GetOptions{})
+	// Create/update lease on managed cluster first. If it fails, it could mean lease resource kind
+	// is not supported on the managed cluster. Create/update lease on the hub then.
+	err := r.updateLease(r.componentNamespace, r.KubeClient)
+
+	if err != nil {
+		glog.Errorf("Failed to update lease %s/%s: %v on managed cluster", r.LeaseName, r.componentNamespace, err)
```

Comment:
We try to recover below, so don't log this error here.  It will be confusing if the recover logic succeeds.

Reply:
I will move it into an else block - as we try to recover under certain conditions.